enumerable.rb

```diff
@@ -0,0 +1,567 @@
+      my_each { |y| return result if y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result unless a.nil? || !a }
+    elsif block_given?
+      my_each { |b| return result if yield(b) }
+    end
+    !result
+  end
+
+  # ==========
+  #  my_none?
+  # ==========
+
+  def my_none?(arg = nil)
+    result = false
+
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result if i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result if x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result if y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result unless a.nil? || !a }
+    elsif block_given?
+      my_each { |b| return result if yield(b) }
+    end
+    !result
+  end
+
+  # ==========
+  #  my_count
+  # ==========
+
+  def my_count(count_one = nil)
+    arr_size = Array(self).length
+    count = 0
+
+    unless block_given?
+      return arr_size if count_one.nil?
+
+      arr_size.times do |i|
+        count += 1 if Array(self)[i] == count_one
+      end
+    end
+
+    if block_given? && !Array(self).nil?
+      arr_size.times do |i|
+        count += 1 if yield(Array(self)[i])
+      end
+    end
+    count
+  end
+
+  # ========
+  #  my_map
+  # ========
+
+  def my_map(proc = nil)
+    return to_enum unless block_given? || proc
+
+    new_arr = []
+    proc ? my_each { |i| new_arr << proc.call(i) } : my_each { |i| new_arr << yield(i) }
+    new_arr
+  end
+
+  # ===========
+  #  my_inject
+  # ===========
+
+  def my_inject(*arg)
+    raise 'LocalJumpError: No block or argument has been given!' if !block_given? && arg[0].nil?
+
+    arr = Array(self)
+
+    arg_one = arg[0]
+    arg_two = arg[1]
+    arg_two = arg_one if arg_two.nil?
+
+    if arg_one.nil? || arg_one.is_a?(Symbol)
+      arr = drop(1)
+      arg_one = to_a[0]
+    else
+      arr = to_a
+    end
+
+    if block_given?
+      arr.my_each do |i|
+        arg_one = yield(arg_one, i)
+      end
+    else
+      arr.my_each do |i|
+        arg_one = arg_one.send(arg_two, i)
+      end
+    end
+    arg_one
+  end
+end
+
+# =============
+#  multiply_els
+# =============
+
+def multiply_els(arr)
+  arr.my_inject(:*)
+end
+
+
+include Enumerable
+array = [2, 7, 5, 9]
+array1 = ['hi', 34, 'potatoes', 'horses', 33]
+array2 = [2, 7, 8, 5]
+hash = { a: 1, b: 2, c: 3, d: 4, e: 5 }
+block = proc { |num| num }
+my_proc = proc { |num| num > 10 }
+range = (5..10)
+true_block = proc { |num| num <= 9 }
+false_block = proc { |num| num >= 9 }
+false_array = [nil, false, nil, false]
+# range = Range.new(5, 50)
+
+puts "\nmy_each output\:"; puts ''
+array1.my_each { |item| puts item }
+puts
+puts "\Array_method_1\:"; puts ''
+p array2.each { |item| item }
+puts
+puts "\Array_method_2(&block)\:"; puts ''
+p array2.my_each(&block)
+puts
+puts "\Range_method_1\:"; puts ''
+range.my_each { |item| puts item }
```

Comment:
- [ ] Kindly delete this. IT is unnecessary for merging. It is ok while you are working on the project, but not for the development branch. You will do better tests later.

Reply:
done